packages/template-retail-react-app/app/components/search/index.jsx

```diff
@@ -116,18 +119,22 @@ const Search = (props) => {
         setSessionJSONItem(RECENT_SEARCH_KEY, searches)
     }
 
-    const debouncedSearch = debounce((input) => searchSuggestion.getSearchSuggestions(input), 300)
+    const debouncedSearch = debounce((input) => {
+        debouncedSearch.cancel()
+        setSearchQuery(input)
+    }, 300)
 
     const onSearchChange = async (e) => {
         const input = e.target.value
         if (input.length >= RECENT_SEARCH_MIN_LENGTH) {
             debouncedSearch(input)
+        } else {
+            setSearchQuery('')
         }
     }
 
     const clearInput = () => {
         searchInputRef.current.blur()
-        searchSuggestion.clearSuggestedSearch()
```

Comment:
Did you forget to reset the `searchQuery` state?

```diff
- searchSuggestion.clearSuggestedSearch()
+ setSearchQuery('')
```

Reply:
Originally I did have the local state being reset, but it was superfluous. I could leave it in there for completeness sake. But didn't seem to be required with the current implementation. But let me take second look at it and think about it a little more.